app/src/main/cpp/emu_jni.cpp

```diff
@@ -85,8 +84,8 @@ extern "C" JNIEXPORT void Java_emu_skyline_EmulationActivity_executeApplication(
     pthread_setname_np(pthread_self(), "EmuMain");
 
     auto jvmManager{std::make_shared<skyline::JvmManager>(env, instance)};
-    auto settings{std::make_shared<skyline::Settings>(preferenceFd)};
-    close(preferenceFd);
+
+    std::shared_ptr<skyline::Settings> settings{std::make_shared<skyline::AndroidSettings>(env, settingsInstance)};
```

Comment:
Auto

Reply:
Auto won't make use of polymorphism though, or is c++ able to upcast automatically?